integration/test_ml.py

```diff
@@ -0,0 +1,389 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for firebase_admin.ml module."""
+import re
+import pytest
+
+
+from firebase_admin import ml
+from firebase_admin import exceptions
+from tests import testutils
+
+
+# pylint: disable=import-error,no-name-in-module
+try:
+    import tensorflow as tf
+    import os      # This is only needed for the tensorflow testing
+    import shutil  # This is only needed for the tensorflow testing
+    _TF_ENABLED = True
+except ImportError:
+    _TF_ENABLED = False
+
+
+@pytest.fixture
+def name_only_model():
+    model = ml.Model(display_name="TestModel123")
+    yield model
+
+
+@pytest.fixture
+def name_and_tags_model():
+    model = ml.Model(display_name="TestModel123_tags", tags=['test_tag123'])
+    yield model
+
+
+@pytest.fixture
+def full_model():
+    tflite_file_name = testutils.resource_filename('model1.tflite')
+    source1 = ml.TFLiteGCSModelSource.from_tflite_model_file(tflite_file_name)
+    format1 = ml.TFLiteFormat(model_source=source1)
+    model = ml.Model(
+        display_name="TestModel123_full",
+        tags=['test_tag567'],
+        model_format=format1)
+    yield model
+
+
+@pytest.fixture
+def invalid_full_model():
+    tflite_file_name = testutils.resource_filename('invalid_model.tflite')
+    source1 = ml.TFLiteGCSModelSource.from_tflite_model_file(tflite_file_name)
+    format1 = ml.TFLiteFormat(model_source=source1)
+    model = ml.Model(
+        display_name="TestModel123_invalid_full",
+        tags=['test_tag890'],
+        model_format=format1)
+    yield model
+
+
+# For rpc errors
+def check_firebase_error(excinfo, status, msg):
+    err = excinfo.value
+    assert isinstance(err, exceptions.FirebaseError)
+    assert err.cause is not None
+    assert err.http_response is not None
+    assert err.http_response.status_code == status
+    assert str(err) == msg
+
+
+# For operation errors
+def check_operation_error(excinfo, msg):
+    err = excinfo.value
+    assert isinstance(err, exceptions.FirebaseError)
+    assert str(err) == msg
+
+
+def _ensure_model_exists(model):
+    # Delete any previously existing model with the same name because
+    # it may be modified from the model that is passed in.
+    _delete_if_exists(model)
+
+    # And recreate using the model passed in
+    created_model = ml.create_model(model=model)
+    return created_model
+
+
+# Use this when you know the model_id and are sure it exists.
+def _clean_up_model(model):
+    ml.delete_model(model.model_id)
+
+
+# Use this when you don't know the model_id or it may not exist.
+def _delete_if_exists(model):
+    filter_str = 'displayName={0}'.format(model.display_name)
+    models_list = ml.list_models(list_filter=filter_str)
+    for mdl in models_list.models:
+        ml.delete_model(mdl.model_id)
+
+
+def test_create_simple_model(name_and_tags_model):
```

Comment:
Once you refactor the fixture as described above, this will just become:

```
def test_create_simple_model(name_and_tags_model):
  # Just the assertions...
  # give the random display name a predictable prefix like: test_model_*, so you can assert for that part
```

Reply:
done